### PR TITLE
Start databases and kafka via docker-compose

### PR DESCRIPTION
**Pull request by Max-Leopold**
Project Board Card: https://github.com/Max-Leopold/mini-analytics/projects/1#card-25671487

### CODE REVIEW

File: Docker/create-collection.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+conn.request("GET", "/solr/admin/collections?action=LIST&wt=json", payload)
+
+res = conn.getresponse()
+data = res.read()
+parsed_json = (json.loads(data))
+
+# Check if collection is already present
+if 'mentions' not in parsed_json['collections']:
+    # Create collection if not present
+    conn.request("GET", "/solr/admin/collections?action=CREATE&name=mentions&numShards=1&replicationFactor=1", payload)
```
Comment (YNedderhoff) [inline Docker/create-collection.py]: Interesting that collections are created with a `GET` request and not `POST`

File: Docker/create-collection.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+conn = http.client.HTTPConnection("solr:8983")
+
+conn.request("GET", "/solr/admin/collections?action=LIST&wt=json", payload)
+
+res = conn.getresponse()
+data = res.read()
+parsed_json = (json.loads(data))
+
+# Check if collection is already present
+if 'mentions' not in parsed_json['collections']:
```
Comment (YNedderhoff) [inline Docker/create-collection.py]: Much nicer than bash ;)

File: Docker/dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM python:3
```
Comment (YNedderhoff) [inline Docker/dockerfile]: Filename should be `Dockerfile` by convention. Folder names should never be uppercase on the other hand, so better `docker`.

File: Docker/dockerfile [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+ADD create-collection.py /create-collection.py
+ADD https://github.com/ufoscout/docker-compose-wait/releases/download/2.5.1/wait /wait
+RUN chmod +x /wait
+CMD /wait &&python /create-collection.py
```
Comment (YNedderhoff) [inline Docker/dockerfile]:
```suggestion
CMD /wait && python /create-collection.py
```

File: Docker/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_DB: QueryDB
+      POSTGRES_USER: admin
+      POSTGRES_PASSWORD: admin
+    volumes:
+      - postgres:/var/lib/postgres/data
+
+  solr:
```
Comment (YNedderhoff) [inline Docker/docker-compose.yml]: I forgot, now that you can create the collection, does cloud mode work with more than one instance?

File: Docker/create-collection.py (hunk 1 above)
Comment (Max-Leopold) [reply Docker/create-collection.py]: Indeed. Didn't realized at the time writing but yes indeed.

File: Docker/docker-compose.yml (hunk 5 above)
Comment (Max-Leopold) [reply Docker/docker-compose.yml]: Didn't try yet.

Comment (YNedderhoff) [reply Docker/docker-compose.yml]: Now that you went through the lengths of getting solr cloud mode to work, that would surely be a rewarding thing :D

Comment (Max-Leopold) [reply Docker/docker-compose.yml]: image

Seems to work.

File: docker/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      WAIT_HOSTS: solr:8983
+      WAIT_HOSTS_TIMEOUT: 60
+
+volumes:
+  kafka1:
+  kafka2:
+  kafka3:
+  zookeeper:
+  postgres:
+  solr:
```
Comment (YNedderhoff) [inline docker/docker-compose.yml]: Is this used?

File: docker/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_DB: QueryDB
+      POSTGRES_USER: admin
+      POSTGRES_PASSWORD: admin
+    volumes:
+      - postgres:/var/lib/postgres/data
+
+  solr:
```
Comment (YNedderhoff) [inline docker/docker-compose.yml]: Maybe make this solr1 to be consistent.

Does the cluster stuff work now?

File: docker/docker-compose.yml (hunk 6 above)
Comment (Max-Leopold) [reply docker/docker-compose.yml]: Now it is

File: Docker/dockerfile (hunk 3 above)
Comment (Max-Leopold) [reply Docker/dockerfile]: Changed

File: docker/docker-compose.yml (hunk 7 above)
Comment (Max-Leopold) [reply docker/docker-compose.yml]: I think so, see https://github.com/Max-Leopold/mini-analytics/pull/25#discussion_r320273684, but I don't know. How could I test it?

**Pull request merged**